cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -61,80 +60,30 @@
+    this.internalAuthEnabled = SecurityUtil.isInternalAuthEnabled(cConf);
   }
 
   @Override
   public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
     throws AccessException {
-    doEnforce(entity, principal, permissions);
-  }
-
-  @Override
-  public void enforceOnParent(EntityType entityType, EntityId parentId, Principal principal, Permission permission)
-    throws AccessException {
-    doEnforce(entityType, parentId, principal, permission);
-  }
-
-  @Override
-  public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal)
-    throws AccessException {
-    // TODO CDAP-17772: Temporary short-circuit, implement internal access enforcement
-    if (principal.getFullCredential() != null && principal.getFullCredential().getType()
-      .equals(Credential.CredentialType.INTERNAL)) {
-      LOG.trace("Internal Principal enforce({}, {})", entityIds, principal);
-      return entityIds;
-    }
-
-    if (!isSecurityAuthorizationEnabled()) {
-      return entityIds;
-    }
-
-    Set<EntityId> visibleEntities = new HashSet<>();
-    // filter out entity id which is in system namespace and principal is the master user
-    for (EntityId entityId : entityIds) {
-      if (isAccessingSystemNSAsMasterUser(entityId, principal) || isEnforcingOnSamePrincipalId(entityId, principal)) {
-        visibleEntities.add(entityId);
-      }
-    }
-
-    principal = getUserPrinciple(principal);
-
-    Set<? extends EntityId> difference = Sets.difference(entityIds, visibleEntities);
-    LOG.trace("Checking visibility of {} for principal {}.", difference, principal);
-    Set<? extends EntityId> moreVisibleEntities;
-    long startTime = System.nanoTime();
-    try {
-      moreVisibleEntities = accessControllerInstantiator.get().isVisible(difference, principal);
-    } finally {
-      long timeTaken = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
-      String logLine = "Checked visibility of {} for principal {}. Time spent in visibility check was {} ms.";
-      if (timeTaken > logTimeTakenAsWarn) {
-        LOG.warn(logLine, difference, principal, timeTaken);
-      } else {
-        LOG.trace(logLine, difference, principal, timeTaken);
-      }
-    }
-    visibleEntities.addAll(moreVisibleEntities);
-    LOG.trace("Getting {} as visible entities", visibleEntities);
-    return Collections.unmodifiableSet(visibleEntities);
-  }
-
-  private void doEnforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
-    throws AccessException {
-    // TODO CDAP-17772: Temporary short-circuit, implement internal access enforcement
-    if (principal.getFullCredential() != null && principal.getFullCredential().getType()
-      .equals(Credential.CredentialType.INTERNAL)) {
+    if (internalAuthEnabled && principal.getFullCredential() != null && principal.getFullCredential().getType()
+      == Credential.CredentialType.INTERNAL) {
```

Comment:
Line break should be at the logical operator.

```
if (internalAuthEnabled && principal.getFullCredential() != null
    && principal.getFullCredential().getType() == Credential.CredentialType.INTERNAL) {
```

Reply:
Since this is a cherry-pick, I've gone ahead and opened a new PR #13533 to address these comments.